beta/nncf/tensorflow/callbacks/statistics_callback.py

```diff
@@ -0,0 +1,68 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+from typing import Callable
+
+import tensorflow as tf
+
+from beta.nncf.helpers.utils import print_statistics
+
+
+class StatisticsCallback(tf.keras.callbacks.Callback):
+    """
+    Callback for logging compression statistics to tensorboard and stdout
+    """
+
+    def __init__(self,
+                 raw_statistics_fn: Callable[[], dict],
+                 log_tensorboard: bool = True,
+                 log_text: bool = True,
+                 log_dir: str = None):
+        """
+        :param raw_statistics_fn: callable to evaluate raw sparsity compression statistics
+        :param log_tensorboard: whether to log statistics to tensorboard or not
+        :param log_text: whether to log statistics to stdout
+        :param log_dir: the directory for tensorbard logging
+        """
+        super().__init__()
+        self._raw_statistics_fn = raw_statistics_fn
+        self._log_tensorboard = log_tensorboard
+        self._log_text = log_text
+        self._file_writer = None
+        if log_tensorboard:
+            if log_dir is None:
+                raise RuntimeError('log_dir must be specified if log_tensorboard is true.')
+            # pylint: disable=no-member
+            self._file_writer = tf.summary.create_file_writer(log_dir + '/compression')
```

Comment:
If we use log_dir only for tensorboard logging, why not log to tensorboard if the path is provided without the additional flag?

Reply:
We've discussed using of explicit `log_tensorboard` flag with @alexsu52 when callback was originally introduced but I don't remember why was chosen this approach (it was long ago), probably for explicitness.